Fix nightly benchmarks

rvl opened the pull request:
### Issue Number

ADP-804

### Overview

Nightly restore bench is still failing.

1. Restore bench: Increase cardano-node startup time allowance

   If the cardano-node database is shared between bench runs, it may need to rebuild the ledger, etc, at startup. So increase the time available for that.

2. Latency bench: Fix working directory

   The meaning of `$src` seems to have changed since the Haskell.nix bump in PR #2533.

3. DB Bench: I broke the criterion env setup - fixed now.

rvl committed:
Restore bench: Increase cardano-node startup time allowance

If the cardano-node database is shared between bench runs, it may need
to rebuild the ledger, etc, at startup. So increase the time available
for that.

KtorZ commented on lib/shelley/bench/Restore.hs:
```diff
@@ -764,7 +764,7 @@ waitForNodeSync
     -> NetworkLayer IO Block
     -> IO SlotNo
-waitForNodeSync tr nw = loop 10
+waitForNodeSync tr nw = loop 120
```
I've seen it took up to 15 minutes locally when switching to newer version of cardano-node. Maybe good to be more conservative here?

rvl replied:
That number works out to 30 minutes, which should be conservative enough, I hope. I added a comment to clarify. It would be clearer if this function used `Control.Retry.retrying` but it doesn't.

piotr-iohk replied:
Hopefully 30 mins is enough (https://input-output-rnd.slack.com/archives/C7PJ29FME/p1616507642042100?thread_ts=1616507577.042000&cid=C7PJ29FME)

rvl replied:
I think on a dedicated linux benchmark machine it should be fine.

rvl committed:
Latency bench: Fix working directory

The meaning of $src seems to have changed since the Haskell.nix bump
in PR #2533.

rvl committed:
DB bench: Fix criterion env setup
